test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+
+    if ($type == 'char' or $type == 'nchar') {
+        // char and nchar may not return the same results - at this point
+        // we've verified that statement execution works so just return
+        return;
+    } elseif ($AEcount > $nonAEcount) {
+        print_r("Too many AE results!\n");
+        print_r($AEres);
+        print_r($nonAEres);
+    } elseif ($AEcount < $nonAEcount) {
+        print_r("Too many non-AE results!\n");
+        print_r($AEres);
+        print_r($nonAEres);
+    } else {
+        if ($AEcount != 0) {
+            $i = 0;
+            foreach ($AEres as $AEr) {
+                if ($AEr[0] != $nonAEres[$i][0]) {
+                    print_r("AE and non-AE results are different! For field $i, got AE result ".$AEres[$i][0]." and non-AE result ".$nonAEres[$i][0]."\n");
+                }
+                ++$i;
+            }
+        }
+    }
+}
+
+// testCompare selects based on a comparison in the WHERE clause and compares
+// the results between encrypted and non-encrypted columns, checking that the
+// results are identical
+function TestCompare($conn, $tableName, $comparisons, $dataTypes, $colNames, $thresholds, $length, $key, $encryptionType, $attestation)
+{
+    foreach ($comparisons as $comparison) {
+        foreach ($dataTypes as $type) {
+
+            // Unicode data types require the PHPTYPE to be specified to UTF-8 string
+            // and the correct collation. Otherwise we get different results between
+            // the encrypted and non-encrypted columns (probably because the collation
+            // was only changed in the encryption query).
+            $string = dataTypeIsStringMax($type);
+            $collate = $string ? " COLLATE Latin1_General_BIN2" : "";
+            $unicode = dataTypeIsUnicode($type);
+            $PDOType = getPDOType($type);
+
+            $AEQuery = "SELECT ".$colNames[$type]."_AE FROM $tableName WHERE ".$colNames[$type]."_AE ".$comparison." ?".$collate;
+            $nonAEQuery = "SELECT ".$colNames[$type]." FROM $tableName WHERE ".$colNames[$type]." ".$comparison." ?".$collate;
+
+            try {
+                $AEstmt = $conn->prepare($AEQuery);
+                $AEstmt->bindParam(1, $thresholds[$type], $PDOType);
+                $nonAEstmt = $conn->prepare($nonAEQuery);
+                $nonAEstmt->bindParam(1, $thresholds[$type], $PDOType);
+            } catch (PDOException $error) {
+                print_r($error);
+                die("Preparing/binding statements for comparison failed");
+            }
+
+            compareResults($AEstmt, $nonAEstmt, $key, $encryptionType, $attestation, $comparison);
+        }
+    }
+}
+
+// testPatternMatch selects based on a pattern in the WHERE clause and compares
+// the results between encrypted and non-encrypted columns, checking that the
+// results are identical
+function TestPatternMatch($conn, $tableName, $patterns, $dataTypes, $colNames, $length, $key, $encryptionType, $attestation)
+{
```

Comment:
Again, please rename the function to be consistent with the one in the comment. Also, add brief descriptions for each argument. See my review comment above